src/mentionediting.js

```diff
@@ -176,3 +217,43 @@ function removePartialMentionPostFixer( writer, doc ) {
 
 	return wasChanged;
 }
+
+// Checks if node has correct mention attribute if present.
+// Returns false if node is text and has mention attribute and its text does not match expected mention text.
+//
+// @param {module:engine/model/node~Node} node a node to check
+// @returns {Boolean}
+function checkMentionAttributeOnNode( node ) {
+	if ( !node || !( node.is( 'text' ) || node.is( 'textProxy' ) ) || !node.hasAttribute( 'mention' ) ) {
+		return true;
+	}
+
+	const text = node.data;
+	const mention = node.getAttribute( 'mention' );
+
+	const expectedText = mention._marker + mention.name;
+
+	return text == expectedText;
+}
+
+// Yields all text nodes with broken mentions from a range - even if mention is outside range boundary.
+//
+// @param {module:engine/range~Range} range
+function* getBrokenMentionsFromRange( range ) {
+	// Check node at the left side of a range.
+	if ( !checkMentionAttributeOnNode( range.start.nodeBefore ) ) {
+		yield range.start.nodeBefore;
+	}
+
+	// Yield text nodes with broken mention from the range.
+	for ( const textProxy of range.getItems() ) {
+		if ( !checkMentionAttributeOnNode( textProxy ) ) {
+			yield textProxy.textNode;
```

Comment:
I understand that this is covered by unit tests but I wonder out of curiosity when this happens?

Reply:
I mean, if that happens for mentions that are not on the boundary of the range. Some something like `[foo<mention />bar]`.

Reply:
So this is used in attribute post-fixer:

```html
fo[o <$text mention="{...}">men]tion</$text>
```

So when the range on which bold is applied is ends inside a mention. Then I want to get that dangling mention node on the right(left) side of range.end(start).

Reply:
So isn't it that we are always interested in nodes next to range boundaries?

Reply:
When applying attribute on this range yes.